deckhouse-controller/pkg/apis/deckhouse.io/v1alpha1/module.go

```diff
@@ -44,41 +92,106 @@ type Module struct {
 }
 
 type ModuleProperties struct {
-	Weight      uint32 `json:"weight"`
-	State       string `json:"state,omitempty"`
-	Source      string `json:"source,omitempty"`
-	Stage       string `json:"stage,omitempty"`
-	Description string `json:"description,omitempty"`
+	Weight           uint32            `json:"weight,omitempty"`
+	Source           string            `json:"source,omitempty"`
+	ReleaseChannel   string            `json:"releaseChannel,omitempty"`
+	Stage            string            `json:"stage,omitempty"`
+	Description      string            `json:"description,omitempty"`
+	Version          string            `json:"version,omitempty"`
+	UpdatePolicy     string            `json:"updatePolicy,omitempty"`
+	AvailableSources []string          `json:"availableSources,omitempty"`
+	Requirements     map[string]string `json:"requirements,omitempty"`
 }
 
 type ModuleStatus struct {
-	Status     string `json:"status"`
-	Message    string `json:"message"`
-	HooksState string `json:"hooksState"`
+	Phase      string            `json:"phase,omitempty"`
+	HooksState string            `json:"hooksState,omitempty"`
+	Conditions []ModuleCondition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
 }
 
-type moduleKind struct{}
-
-func (in *ModuleStatus) GetObjectKind() schema.ObjectKind {
-	return &moduleKind{}
+type ModuleCondition struct {
```

Comment:
Why not to use [apimachinery condition](https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L1573), to use [this](https://github.com/kubernetes/apimachinery/blob/master/pkg/api/meta/conditions.go#L31) method